core/src/test/java/org/apache/accumulo/core/spi/compaction/DefaultCompactionPlannerTest.java

```diff
@@ -302,6 +317,60 @@ public void testErrorExternalNoQueue() {
     assertTrue(e.getMessage().contains("queue"), "Error message didn't contain queue");
   }
 
+  /**
+   * Tests queue with missing name throws error
+   */
+  @Test
+  public void testErrorQueueNoName() {
+    DefaultCompactionPlanner planner = new DefaultCompactionPlanner();
+    Configuration conf = EasyMock.createMock(Configuration.class);
+    EasyMock.expect(conf.isSet(EasyMock.anyString())).andReturn(false).anyTimes();
+
+    ServiceEnvironment senv = EasyMock.createMock(ServiceEnvironment.class);
+    EasyMock.expect(senv.getConfiguration()).andReturn(conf).anyTimes();
+    EasyMock.replay(conf, senv);
+
+    String queues =
+        "[{\"name\":\"smallQueue\", \"maxSize\":\"32M\"}, {\"type\":\"internal\", \"queue\":\"broken\"}]";
```

Comment:
That's a good point. I'll make this test more explicit.

Reply:
I made the test more explicit and also added additional json parsing logic so the code errors on additional fields instead of just silently ignoring them.